### ISSUE TITLE
i18n.merge_file() + build_by_default = warning

### ISSUE THREAD

**Issue by GeorgesStavracas** [mesonbuild/meson#5079]
Meson 0.49.2 throws the following warnings when building GNOME Settings:

```
tests/interactive-panels/applications/meson.build:14: WARNING: Passed invalid keyword argument "build_by_default".
WARNING: This will become a hard error in the future.
```
Here's the snippet that triggers it:

```
desktop_files = [
  'dynamic-panel',
  'header-widget',
  'sidebar-widget',
  'static-init',
  'toplevel-sidebar-widget',
]

foreach desktop_file : desktop_files
  i18n.merge_file(
              desktop,
                type : 'desktop',
               input : 'gtp-@0@.desktop.in'.format(desktop_file),
              output : 'gnome-@0@-panel.desktop'.format(desktop_file),
              po_dir : po_dir,
    build_by_default : true,
             install : false,
  )
endforeach
```

Discussing on IRC, this is a buggy behavior, since i18n actually uses the `build_by_default` value, and yet warns about it.

**Comment by dcbaker** [comment]
@nirbheek, this isn't a 0.50 regression, but how would you feel about adding the fix to 0.50.1?

**Comment by nirbheek** [comment]
> @nirbheek, this isn't a 0.50 regression, but how would you feel about adding the fix to 0.50.1?

Looks fine to me!
